lib/greed/game.rb

```diff
@@ -0,0 +1,102 @@
+absolute_path = File.expand_path(File.dirname(__FILE__))
+require absolute_path + '/turn'
+require absolute_path + '/player'
+
+module Greed
+  class Game
+
+    def initialize
+      @turn = Turn.new
+      @players = []
+    end
+
+    def new_game
+      player_count = player_count?
+      init_players(player_count)
+      get_player_names
+      play_game
+    end
+    
+    # Complete a full game of Greed, entering a final round when any player reaches
+    # a score of 3000 or more
+    def play_game
+      final_round = false
+      while not final_round
+        current_player = @players.pop
+        puts "Current player: #{current_player.name} (current score: #{current_player.score})"
+        score = @turn.take_turn
+
+        # Add score to player total if the player is 'in the game'
+        # (scored more than 300 in a single turn at least once)
+        if current_player.score > 0 || score >= 300
+          puts "Your final score for this round was #{score}"
+          current_player.score += score
+          puts "Your total score is now #{current_player.score}"
+        else
+          puts "You did not score high enough to enter the game."
+        end
+        puts # Spacing
+
+        # Check if this player has reached 3000, which triggers a final round
+        if current_player.score >= 3000 then final_round = true end
```

Comment:
Exercise a lot of caution with imperative assignment inside a block scope. Although there are no issues in this case it can be easy to do this but then leave some code paths open to an undefined variable exception. 

A better alternative it to lift the assignment to scope above,

```ruby
final_round = 
  if current_player.score >= 3000
    true
  else
    false
  end
```

 refactor to not rely on an assignment

```ruby
 break if current_player.score >= 3000
```

(the last recommendation also mean you get to drop the `if not` conditional on the next row)